flag/guest/cluster/kubernetes/kubernetes.go

```diff
@@ -0,0 +1,65 @@
+package kubernetes
+
+// Kubernetes is a data structure to hold guest cluster Kubernetes specific
+// configuration flags.
+type Kubernetes struct {
+	API               API
+	Domain            string
+	Hyperkube         Hyperkube
+	IngressController IngressController
+	Kubelet           Kubelet
+	NetworkSetup      NetworkSetup
+	SSH               SSH
+}
+
+// API is a data structure to hold guest cluster Kubernetes API specific
+// configuration flags.
+type API struct {
+	AltNames       string
+	ClusterIPRange string
+	InsecurePort   string
+	SecurePort     string
+}
+
+// Hyperkube is a data structure to hold guest cluster Kubernetes Hyperkube
+// image specific configuration flags.
+type Hyperkube struct {
+	Docker Docker
+}
+
+// IngressController is a data structure to hold guest cluster ingress
+// controller specific configuration flags.
+type IngressController struct {
+	Docker Docker
+}
+
+// Kubelet is a data structure to hold guest cluster kubelet specific
+// configuration flags.
+type Kubelet struct {
+	AltNames string
+	Labels   string
+	Port     string
+}
+
+// NetworkSetup is a data structure to hold guest cluster network setup
+// configuration flags.
+type NetworkSetup struct {
+	Docker Docker
+}
+
+// Docker is a data structure to hold Docker image configuration flag.
+type Docker struct {
+	Image string
+}
+
+// SSH is a data structure to hold guest cluster SSH specific configuration
+// flags.
+type SSH struct {
+	UserList SSHUser
```

Comment:
Basically these are just keys, but is this structure correct?

Reply:
I don't think a list will work like this. I believe we usually concat lists with commas, so we have a single string which we then split. This was the only example I could think of: https://github.com/giantswarm/kubernetesd/search?utf8=%E2%9C%93&q=AltNames&type=

I am not sure if this is still the way to go, but I remember there was also one case where we did this with the string being formatted as `user1:pubkey1,user2:pubkey2` and so on. 

So if that is still correct then it should be `UserList string`, let's wait for one of the other guys to confirm/deny though.

Reply:
I don't think viper or microkit does lists at all in config files currently, which is why we serialize lists (e.g. the projectlist in installation that you already edited which is used by Draughtsman). Again, I am not 100% about this.